sdk/src/main/scala/io/horizen/storage/AbstractHistoryStorage.scala

```diff
@@ -141,6 +139,31 @@ abstract class AbstractHistoryStorage[
     sidechainBlockInfo.mainchainHeaderBaseInfo.last
   }
 
+  def getNumOfScBlocksWithNoMcRefDataSince(blockId: ModifierId): Int = {
```

Comment:
also: let's add an extra check at the beginning and in the loop: blockId != 0x0000000000.....0000
(it means that we reached the genesys block: can't happen in EON since the fork is activated after, but may be theoretically possible to have a new chain with the fork active from block 1)

Reply:
Ok done